base/dates/io.jl

```diff
@@ -71,7 +71,7 @@ function DateFormat(f::AbstractString,locale::AbstractString="english")
     for (i,k) in enumerate(s)
         k == "" && break
         tran = i >= endof(s) ? r"$" : match(Regex("(?<=$(s[i])).*(?=$(s[i+1]))"),f).match
-        slot = tran == "" || tran == r"$" ? FixedWidthSlot : DelimitedSlot
+        slot = tran == "" ? FixedWidthSlot : DelimitedSlot
```

Comment:
I think so. If `tran <: Regex`, `slot` must be a `DelimitedSlot` for the correct `getslot` method to be invoked. It was working till now because the equality test on Regex types was failing.
@quinnj ?